docs/access_and_permissions.md

```diff
@@ -244,6 +299,61 @@ flowchart TD
     G --Pundit.policy_scope(current_user, SupportedPermission)--> H("SupportedPermissionPolicy (scope)")
 ```
 
+#### Users signin permissions create
+
+These dependencies determine whether a user can:
+
+- complete the controller action
+- grant a given user access to an app
+
+```mermaid
+flowchart TD
+    A(Users::SigninPermissionsController#create)
+    --authorize [application, user: @user], :grant_signin_permission?, policy_class: Users::ApplicationPolicy
+    --> B(Users::ApplicationPolicy#grant_signin_permission?)
+
+    A --"UserUpdate.new(@user, { supported_permission_ids: }, current_user, user_ip_address).call"
+    --> C(UserUpdate#call)
+
+    C --SupportedPermissionParameterFilter.new(current_user, user, user_params) [...] .filtered_supported_permission_ids
+    --> D(SupportedPermissionParameterFilter#filtered_supported_permission_ids)
+    D --Pundit.policy_scope(current_user, SupportedPermission)--> E("SupportedPermissionPolicy (scope)")
+```
+
+#### Users signin permissions delete
+
+These dependencies determine whether a user can:
+
+- view the confirmation screen for revoking access to a given app
+
+```mermaid
+flowchart TD
+    A(Users::SigninPermissionsController#delete)
+    --authorize [application: @application, user: @user], :remove_signin_permission?, policy_class: Users::ApplicationPolicy
+    --> B(Users::ApplicationPolicy#remove_signin_permission?)
+```
+
+#### Users signin permissions destroy
+
+These dependencies determine whether a user can:
+
+- complete the controller action
+- revoke access to a given app
+
+```mermaid
+flowchart TD
+    A(Users::SigninPermissionsController#destroy)
+    --authorize [application, user: @user], :remove_signin_permission?, policy_class: Users::ApplicationPolicy
+    --> B(Users::ApplicationPolicy#remove_signin_permission?)
+
+    A --"UserUpdate.new(@user, { supported_permission_ids: }, current_user, user_ip_address)"
+    --> C(UserUpdate#call)
+
+    C --SupportedPermissionParameterFilter.new(current_user, user, user_params) [...].filtered_supported_permission_ids
```

Comment:
And (minor!) removed the space after `[...]` which is present after the other uses of that ellipsis